app/models/pawn.rb

```diff
@@ -1,11 +1,42 @@
 class Pawn < Piece
 
+  def valid_move?(x, y)
+    unless super(x, y)
+      return false
+    end
+    x_distance = x_position - x
+    y_distance = y_position - y
+    if occupied?(x, y)
+      if color == 'white'
+        x_distance.abs == 1 && y_distance == 1
+      else
+        (-x_distance).abs == 1 && -y_distance == 1
+      end
+    elsif color == 'white'
+      unless has_moved
+        (x_distance == 0 && y_distance == 1) || (x_distance == 0 && y_distance == 2)
+      else
+        x_distance == 0 && y_distance == 1
+      end
+    else
+      unless has_moved
+        (-x_distance == 0 && -y_distance == 1) || (-x_distance == 0 && -y_distance == 2)
+      else
+        -x_distance == 0 && -y_distance == 1
+      end
+    end
+  end
+
+  def occupied?(x, y)
```

Comment:
https://github.com/queens-pawn/chess_app/blob/master/app/models/piece.rb#L13-L23

`move_to` already checks color of the piece, which similar to what you're doing with `occupied?`

can you think of a way to refactor `move_to!` and pull the logic you're looking for?